### PR TITLE
Feature/add compiler feature

### PR DESCRIPTION
**Pull request by tbrezot**
Enable importing one algorithm only from CryptoCore. This will be useful not to compile asymmetric algorithms when compiling Findex for example.

### CODE REVIEW

**Comment by ThibaudDauce** [comment]
> This will be useful

Is `crypto_core` slow to compile? Or does the tree shaker of Rust not good enough to remove unused functions inside the final binary?

**Comment by tbrezot** [comment]
Compilation of Findex with the v7.1:
```
real    0m12.705s
user    0m17.387s
sys     0m3.549s
```

Compilation of Findex with the featured version:
```
real    0m5.432s               
user    0m12.992s                      
sys     0m2.768s                                                                     
```

Which is 22% better (based on the system time).

**Review by Manuthor — DISMISSED** [review]
great!

File: examples/examples/symmetric_crypto.rs [hunk 1]
```diff
@@ -3,13 +3,13 @@
 /// Demonstrates how to use symmetric encryption in combined mode
 /// where
 ///  - the plaintext is a single vector of bytes and
-///  - the ciphertext is a newly allocated vector
-///    that combines the encrypted data and the MAC
+///  - the ciphertext is a newly allocated vector that combines the encrypted
+///    data and the MAC
+#[cfg(all(feature = "chacha"))]
```
Comment (ThibsG) [inline examples/examples/symmetric_crypto.rs:8]:
```suggestion
#[cfg(feature = "chacha")]
```

File: examples/examples/symmetric_crypto.rs [hunk 2]
```diff
@@ -61,12 +63,12 @@ pub fn dem_vector_combined() {
...
```
Comment (ThibsG) [inline examples/examples/symmetric_crypto.rs:68]:
```suggestion
#[cfg(feature = "chacha")]
```

File: examples/examples/symmetric_crypto.rs [hunk 3]
```diff
@@ -124,11 +129,12 @@ pub fn dem_vector_detached() {
 
 /// Demonstrates how to use symmetric encryption
 /// with a stream of data
+#[cfg(all(feature = "chacha"))]
```
Comment (ThibsG) [inline examples/examples/symmetric_crypto.rs:132]:
```suggestion
#[cfg(feature = "chacha")]
```

File: examples/main.rs [hunk 4]
```diff
@@ -1,17 +1,34 @@
...
+        feature = "blake",
+        feature = "curve25519"
+    ))]
+    self::examples::ecies_x25519_xchacha20_stream();
+
+    #[cfg(all(feature = "curve25519"))]
+    self::examples::ed25519_static();
+    #[cfg(all(feature = "curve25519"))]
+    self::examples::ed25519_cached();
+    #[cfg(all(feature = "curve25519"))]
```
Comment (ThibsG) [inline examples/main.rs:32]:
```suggestion
    #[cfg(feature = "curve25519")]
    self::examples::ed25519_static();
    #[cfg(feature = "curve25519")]
    self::examples::ed25519_cached();
    #[cfg(feature = "curve25519")]
```

**Review by ThibsG — COMMENTED** [review]
One idea would be to use [`cargo hack`](https://github.com/taiki-e/cargo-hack) with the `--feature-powerset` option to check compilation and clippy for any combination of features.

Also as discussed, a `default` with everything enabled would be great 👍🏻

**Review by Manuthor — APPROVED** [review]

**Pull request merged**